src/app/modules/form/form.service.ts

```diff
@@ -348,44 +350,74 @@ export const retrievePublicFormsWithSmsVerification = (
 
 export const createSingleSampleSubmissionAnswer = (field: FormFieldDto) => {
   let sampleValue = null
-  let noOfOptions = 0
-  let randomSelectedOption = 0
+  let noOfTableRows
+  let noOfTableCols
+  const tableSampleValue = []
   switch (field.fieldType) {
-    case 'textarea':
-    case 'textfield':
-      sampleValue =
-        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
+    case BasicField.LongText:
+      sampleValue = faker.lorem.sentence()
       break
-    case 'radiobutton':
-    case 'dropdown':
-      noOfOptions = field.fieldOptions.length
-      randomSelectedOption = Math.floor(Math.random() * noOfOptions)
-      sampleValue = field.fieldOptions[randomSelectedOption]
+    case BasicField.ShortText:
+      sampleValue = faker.lorem.word()
       break
-    case 'email':
-      sampleValue = 'hello@example.com'
+    case BasicField.Radio:
+    case BasicField.Dropdown:
+      sampleValue = faker.helpers.arrayElement(field.fieldOptions)
       break
-    case 'decimal':
-      sampleValue = 1.234
+    case BasicField.Email:
+      sampleValue = faker.internet.email()
       break
-    case 'number':
-      sampleValue = 1234
+    case BasicField.Decimal:
+      sampleValue = faker.number.float({ precision: 0.1 }).toString()
       break
-    case 'mobile':
-      sampleValue = '+6598765432'
+    case BasicField.Number:
+      sampleValue = faker.number.int(100).toString()
       break
-    case 'homeno':
-      sampleValue = '+6567890123'
+    case BasicField.Mobile:
+      sampleValue = faker.phone.number('+659#######')
       break
-    case 'yes_no':
-      sampleValue = 'yes'
+    case BasicField.HomeNo:
+      sampleValue = faker.phone.number('+656#######')
       break
-    case 'rating':
-      sampleValue = 1
+    case BasicField.YesNo:
+      sampleValue = faker.helpers.arrayElement(['Yes', 'No'])
```

Comment:
Maybe this should be separate issue: Feels somewhat odd that we can use `field.fieldOptions` for radio and dropdown options, but have to resort to hard-coding `'Yes', 'No'` for this when these options are determined on our end already. Maybe we should have like something like this:
```ts
// in a const file somewhere
export const yesNoOptions = ['Yes', 'No'] as const

// in frontend/src/templates/Field/types.ts
export type YesNoFieldValue = typeof yesNoOptions[number]
```
Then we can just import `yesNoOptions` in places like these where we want to use the options.

Reply:
Yeah this is a good point! Yes and No are also being hardcoded in the options now for the YesNo field. I'll leave it as is for now, but will create a separate issue to improve this

Reply:
added here https://linear.app/ogp/issue/FRM-892/refactor-yes-no-field-options